Src/Newtonsoft.Json/Utilities/MiscellaneousUtils.cs

```diff
@@ -39,15 +39,11 @@ internal static class MiscellaneousUtils
     {
         public static bool ValueEquals(object objA, object objB)
         {
-            if (objA == null && objB == null)
+            if (objA == objB)
             {
                 return true;
             }
-            if (objA != null && objB == null)
-            {
-                return false;
-            }
-            if (objA == null && objB != null)
+            if (objA == null | objB == null)
```

Comment:
Null checks against locals are fast, so the cost of an extra check when objA is null is worth it to have one less branch.

Its something that puts the micro in micro-opt, but if I have to choose between two options, might as well be the slightly quicker.